### PR TITLE
Update the Azure implicit authentication docs

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-8328, DOC-8371, DOC-8051, DOC-8053

This PR updates the Azure Implicit Auth section of the docs to use the new credentials file that CockroachDB will check on chain lookup.

This is new to v23.2 and v23.1.9.

### CODE REVIEW

**Comment by kathancox** [comment]
Hey @adityamaru, appreciate if you could review this one too, since you reviewed the original eng PR. I have updated the Azure implicit auth section to just mention the credentials file.... I removed mention that users could store the tenant ID etc. individually as env vars. I think that users could still do this, but I assumed the new creds file way was more efficient. Let me know if you think I should re-include the individual vars.

**Comment by adityamaru** [comment]
>  Let me know if you think I should re-include the individual vars.

For the sake of existing setups I _think_ we should keep what we had before and add this file based azure configuration as an additional option (maybe the preferred one if we want, but I don't feel strongly about this). In code we first check the file, then env vars, then managed identity. So I think keeping the existing literature in the docs will reflect that fact more clearly.

**Comment by kathancox** [comment]
> > Let me know if you think I should re-include the individual vars.
> 
> For the sake of existing setups I _think_ we should keep what we had before and add this file based azure configuration as an additional option (maybe the preferred one if we want, but I don't feel strongly about this). In code we first check the file, then env vars, then managed identity. So I think keeping the existing literature in the docs will reflect that fact more clearly.

@adityamaru Got it, I have added in the three options. Let me know what you think! Thank you

**Review by adityamaru — APPROVED** [review]

File: src/current/v23.1/cloud-storage-authentication.md [hunk 1]
```diff
@@ -651,23 +651,82 @@ You can authenticate to Azure with explicit credentials in the following ways:
...
-- `AZURE_TENANT_ID`: Directory (tenant) ID for your App Registration.
+- A credentials file with the path specified in the environment variable `COCKROACH_AZURE_APPLICATION_CREDENTIALS_FILE`. Refer to [Set up a credentials file](#set-up-a-credentials-file).
+- Each credential set as an environment variable.
+    - `AZURE_CLIENT_ID`: Application (client) ID for your [App Registration](https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app#register-an-application).
+    - `AZURE_CLIENT_SECRET`: Client credentials secret generated for your App Registration.
+    - `AZURE_TENANT_ID`: Directory (tenant) ID for your App Registration.
+- An Azure [managed identity](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/overview).
 
-See Microsoft's [Azure Authentication](https://learn.microsoft.com/azure/developer/go/azure-sdk-authentication) documentation to set this up. 
+You can use this for authenticating an Azure Blob Storage URI or an Azure Key Vault URI for an [Azure encrypted backup]({% link {{ page.version.version }}/take-and-restore-encrypted-backups.md %}).
```
Comment (florence-crl) [inline src/current/v23.1/cloud-storage-authentication.md:663]: Maybe modify the link to go to this specific section: https://deploy-preview-18119--cockroachdb-docs.netlify.app/docs/v23.1/take-and-restore-encrypted-backups#azure-key-vault-uri-format

File: src/current/v23.1/cloud-storage-authentication.md [hunk 2]
```diff
@@ -651,23 +651,82 @@ You can authenticate to Azure with explicit credentials in the following ways:
...
 
-You must include the container name and Azure account name in your URI, as follows:
+{{site.data.alerts.callout_info}}
+You must export the environment variable on each CockroachDB node.
```
Comment (florence-crl) [inline src/current/v23.1/cloud-storage-authentication.md:666]: Does this note only apply to bullet 2 regarding environmental variables? If so, should this note be after line 660?

Comment (florence-crl) [reply src/current/v23.1/cloud-storage-authentication.md:666]: From line 691, I realized that the note also applies to bullet 1. Does it apply to bullet 3 as well?

**Review by florence-crl — APPROVED** [review]
LGTM with minor feedback regarding positioning of a note and modifiying a link.

File: src/current/v23.1/cloud-storage-authentication.md (hunk 1 above)
Comment (kathancox) [reply src/current/v23.1/cloud-storage-authentication.md]: Updated these links here. Thanks!

**Pull request merged**